Sources/StreamChatUI/ChatThreadList/ChatThreadListItemView.swift

```diff
@@ -0,0 +1,264 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// The view that displays the thread information in the Thread List.
+///
+/// Default layout:
+/// ```
+/// |----------------------------------------------------------|
+/// | icon | threadTitleLabel                                  |
+/// | ---------------------------------------------------------|
+/// | threadDescriptionLabel                 | unreadCountView |
+/// |----------------------------------------------------------|
+/// |            | replyTitleLabel                             |
+/// | avatarView | --------------------------------------------|
+/// |            | replyDescriptionLabel      | timestampLabel |
+/// |----------------------------------------------------------|
```

Comment:
This looks simple, but in fact, there are many hidden containers here. I thought about splitting this view into 2, the `threadContainer` would a view and the `replyContainer` another view. This would make the code cleaner, but less customizable. It would not be possible to move views from one of the container to another one. This is why also I started thinking about the new Container Result Builder to improve this.